discord/client.py

```diff
@@ -610,8 +610,8 @@ async def login(self, token: str) -> None:
         token = token.strip()
 
         data = await self.http.static_login(token)
-        self._connection.user = ClientUser(state=self._connection, data=data)
-        self._application = await self.application_info()
+        self._connection.user = ClientUser(state=self._connection, data=data['bot'])  # type: ignore
```

Comment:
Not sure if it would be better to raise above here if `'bot' not in data`.
I don't think it can ever happen though.